### DISCUSSION TITLE
Reconcile source time functiosn for CMTSOLUTION and FORCESOLUTION

### DISCUSSION THREAD

**Discussion by homnath** [SPECFEM/specfem3d#1651] [SPECFEM3D General]
@SPECFEM/specfem-admins Currently, we have greater flexibility in selecting the source time function for FORCESOLTUON but not CMTSOLUTION, mainly due to CMTSOLUTION standard format. What is the best strategy to bring other source time functions, such as newly implemented Brune and smooth Brune functions, to CMTSOLUTION? Any suggestion would be greatly appreciated :)

Best,
Hom Nath

**Comment by elifo** [comment]
I'd say that as long as the spectral amplitude of Brune's function equals 1  at low frequencies (the plateau), you would just need to multiply the STF with moment tensor. This is exactly what the code does when external source time functions are used. 

Your implementation seems to satisfy that:
<img width="645" alt="STF1" src="https://github.com/SPECFEM/specfem3d/assets/26336068/6937517f-c0a8-4c18-97c5-64aad26c9846">
<img width="483" alt="STF2" src="https://github.com/SPECFEM/specfem3d/assets/26336068/8ca84027-9051-4e2b-8157-a33a3703b24f">

**Reply by homnath** [reply]
Thank you so much @elifo ! We want to avoid using external source files due to a massive number of source points.

**Reply by elifo** [reply]
(Oh I had meant that you would only need to do the same thing as external STFs, when implementing Brune for CMTSOLUTION use.) 
Thank you very much for the implementation which will be of great use to me as well:)

**Reply by homnath** [reply]
Thanks! Welcome!

**Comment by jpampuero** [comment]
I would add an option USE_SOURCE_TIME_FUNCTION,
and optional lines at the bottom of the CMTSOLUTION file to set the additional STF parameters, if needed.

**Reply by homnath** [reply]
Nice idea! Thanks!

**Comment by homnath** [comment]
In my view, the best way to reconcile source time functions for CMTSOLUTION and FORCESOLUTION is to create a new common parameter for both CMTSOLUTION and FORCESOLUTION similar to:
SOURCE_TIME_FUNCTION = gaussian
and get rid of USE_RICKER_TIME_FUNCTION from the Par_file and "source time function" from the FORCESOLUTION.

However, to minimise the code changes, I have added a new parameter, USE_OTHER_TIME_FUNCTION, for now. Currently, it is implemented only for viscoelastic simulations, and the other available functions are Brune and Smooth Brune.
